tests/kernels/test_gguf.py

```diff
@@ -0,0 +1,112 @@
+from pathlib import Path
+from typing import List
+
+import pytest
+import torch
+from gguf import GGMLQuantizationType, GGUFReader, ReaderTensor, dequantize
+from huggingface_hub import snapshot_download
+
+import vllm._custom_ops as ops
+
+GGUF_SAMPLE = snapshot_download("Isotr0py/test-gguf-sample")
+
+
+def get_gguf_sample_tensors(
+        hidden_size: int,
+        quant_type: GGMLQuantizationType) -> List[ReaderTensor]:
+    sample_dir = GGUF_SAMPLE
+    filename = f"Quant_{quant_type.name}_{hidden_size}.gguf"
+    sample_file = Path(sample_dir) / filename
+    return GGUFReader(sample_file).tensors
```

Comment:
The `gguf-py` only implement quantize method for `Q4_0`, `Q5_0` and `Q8_0` in v0.10.0, remaining the others unimplemented. 

So, I have created these sample GGUF files for kernel test through C bindings. The creation script also available in the repo: [create_sample.py](https://huggingface.co/Isotr0py/test-gguf-sample/blob/main/create_sample.py)

Reply:
This is a great idea, thanks a ton